okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/ConcurrencyLimiters.java

```diff
@@ -333,7 +333,49 @@ public void onDropped() {
                 }
             };
             leakDetector.register(result, allocationStackTrace);
-            return result;
+            return new AtMostOneInteractionListener(result);
+        }
+    }
+
+    /** Prevent multiple interactions with the same listener from modifying state in unexpected ways. */
+    @VisibleForTesting
+    static final class AtMostOneInteractionListener implements Limiter.Listener {
+
+        private final Limiter.Listener delegate;
+        private boolean completed;
+
+        AtMostOneInteractionListener(Limiter.Listener delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public void onSuccess() {
```

Comment:
Do you need some synchronization here?

Reply:
No, limiters are not shared between threads so we don't need atomic/volatile variables. There's a happens-before relationship.

Reply:
Yeah, I had a look and it seemed that way, but better ask.